datalad/utils.py

```diff
@@ -906,12 +982,17 @@ def unique(seq, key=None, reverse=False):
       If True, uniqueness checked in the reverse order, so that the later ones
       will take the order
     """
-    seen = set()
+    seen: set[T] = set()
     seen_add = seen.add
 
-    trans = reversed if reverse else lambda x: x
+    if reverse:
+        def trans(x: Sequence[T]) -> Iterable[T]:
+            return reversed(x)
+    else:
+        def trans(x: Sequence[T]) -> Iterable[T]:
+            return x
 
-    if not key:
+    if key is None:
```

Comment:
curious: is such change needed for type annotation as otherwise it cannot figure out something?

Why it is the right thing to do in general, I think: I guess in principle there could be a callable instance of a class which has some `__bool__` which would evaluate it to False and thus this check would be incorrectly handled here. Is that the "functional" (or may be what type checking potentially complains) reason @jwodder ?

Reply:
I don't think this was done for typing reasons; I think I just let my dislike of that kind of sloppy conditional (If you *mean* `is None`, *say* `is None`) get away from me for a bit.